File: C++/Comments/Array/238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans;
        int prefixL = 1;
        for (int n : nums) {
            prefixL *= n;
            ans.push_back(prefixL);
        }
        int prefixR = 1;
        for (int i = nums.size() - 1; i > 0; i--) {
            ans[i] = prefixR * ans[i - 1];
            prefixR *= nums[i];
        }
        ans[0] = prefixR;
        return ans;
    }
};

File: Python/238.product_of_array_except_self.py
from typing import List
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        running = 1
        ans = []
        for num in nums:
            running *= num
            ans.append(running)
        running_r = 1
        N = len(nums)
        for i in range(N-1, 0, -1):
            ans[i] = running_r * ans[i-1]
            running_r *= nums[i]
        ans[0] = running_r
        return ans